### ISSUE TITLE
Midpoint Review - Accessibility Feedback - VCL Seamless Call Solution

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#90128]
## Next Steps for the VFS team

- [ ] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [ ] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- Really appreciate the work you're doing here. Always good to provide more ways to contact the VCL.
- In the meeting I gave a book recommendation but I misremembered the title! The correct title is [*Design for Real Life*](https://www.sarawb.com/design-for-real-life), by Eric Meyer and Sara Wachter-Boettcher. (I was remembering "Designing for Crisis," an article and conference talk by Meyer that later became part of the book.) Can't emphasize enough the importance of thinking about how stress impacts a user's ability to read and interpret instructions, understand a task, etc.
- If you haven't already, I definitely encourage you to connect with the folks in the [#trauma-practice channel in Slack](https://dsva.slack.com/archives/C04F9JLSATE). There are a lot of thoughtful people in there who can help you explore UI ideas or plan for any future research.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [ ] **Must:** It sounds like this is already accounted for, but definitely make sure everything renders as intended on mobile devices. Mobile accounts for a huge part of VA.gov traffic, and a Veteran experiencing a crisis may not always be sitting in front of a desktop.
- [ ] **Must:** Each button needs to have a visible label. There are a few reasons why, but I'll call out two kinds of user interaction specifically:
   - Sighted voice command/speech recognition users often speak commands like "Click Mute," which then clicks the mute button. Their choice of what words to say is informed by the text they read on the screen. If there's an unlabeled button, they have to guess what the magic word is to click it --- eg. for the green button that starts the call, "Click Call" or "Click Start Call" or "Click Phone" etc.  Voice software typically provides multiple ways to reach a button or link on the page, but including visible text reduces that point of potential friction.
   - If a user is on a call with the VCL and can't provide their location, the VCL rep is going to have a much easier time saying "Do you see a button that says 'Share location'? Click that button" than trying to describe the map marker icon. Again, it's not an impossible interaction, but it's an opportunity to reduce friction that doesn't have to be there.
- [ ] **Should:** Thinking about the location and settings buttons, I'd encourage you to look at the [Button Icon](https://design.va.gov/components/button/button-icon) component in the design system. The component itself doesn't support much customization, but I think that's a good model to follow for consistency with other elements on VA.gov.
- [ ] **Must:** I can't tell for certain from the prototype, but the text size for the button labels (Mute and Keypad) seems a little on the small side. Note that almost all of the rest of VA.gov is at a text size of 16.96px or larger, and it doesn't make sense for these elements to be harder to read than regular body copy.
- [ ] **Must:** From the Figma file, it's not clear what the focus states for the buttons will be. I just want to emphasize the importance of having the keyboard focus indicator visible when a button receives focus, and that the focus indicator contains all elements of the button. I'm thinking of the three primary buttons (mute, start/end call, keypad). The focus indicator needs to wrap both the circular button and the text that goes with it.
- [ ] **Must:** In general, VA.gov avoids floating UI elements to the right without any other visual indicator that they're out there. The big concern here is low vision users who use screen magnification software or magnifying devices. Once a magnification user finds the main column of content on a page, they don't typically go looking for things in the empty white space around that column, so elements floated to the left or right are unlikely to be found. I'd encourage you to move the location and settings buttons into the center column.
- [ ] **Coding note:** The Call Settings box that toggles open when you click the settings button should have the same accessibility features as a modal --- namely, it should close when pressing [esc], it should have a close button in the top-right corner, keyboard focus should move to content inside the box when it opens, and focus should be trapped inside the box while it's open. Easiest solution here is to just use the `va-modal` component, but if that's not an option then definitely use `va-modal` as a template for keyboard behavior.
- [ ] **Should:** In the Call Settings box, I would strongly recommend that there's a "Save Settings" button (or named something similar) that applies changes to your microphone and speaker settings, rather than just the dropdowns. I don't think [WCAG 3.2.2](https://www.w3.org/WAI/WCAG22/Understanding/on-input.html) technically applies here since there's no change in context, but part of the motivation for 3.2.2 is that it's relatively easy for keyboard users to accidentally change their dropdown selection without realizing it, especially if they're also screen reader users not receiving visual feedback about changes. It also simplifies the modal-like interaction model --- choose your settings, click save, the box closes and keyboard focus moves back to the Settings button.
- [ ] **Should:** I didn't think to mention this in the meeting, but I'd encourage you to think about the order of elements for screen reader users. Screen reader users often experience a page linearly, where each element of the page occurs in sequence, usually top-to-bottom. As currently presented, the order reads to me as:
   - H1: Call the Veterans Crisis Line over the Internet
   - Call the VA from your web browser...
   - Button: Location
   - Button: Settings
   - End-to-end encrypted
   - Veterans Crisis Line
   - Button: Mute
   - Button: Start call
   - Button: Keypad
   
   The primary action of the page is buried pretty deep in that sequence. Sighted Veterans will typically be able to identify the call button and jump straight there, but a blind Veteran who doesn't get that visual cue is going to have to get through a lot of stuff before they get to the call button. 
   
   I realize this layout mimics a phone interface, but I think there's space to explore other layouts that approach the sequence of elements on the page in a more deliberate way. If you took away all of the visuals and were just navigating this page as text, what's the sequence of elements that makes the most sense?
- [ ] **Coding note:** Thinking about screen reader users again, make sure the status of the call (calling, call started, call ended, and any other statuses that may apply) is communicated in real time via screen readers. There are a few approaches to doing this, but I think an [aria-live](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live) region makes the most sense here.
- [ ] **Must:** In the prototype, we didn't see anything relating to error states, eg. a call failed to connect. Make sure those states are accounted for, and that they're communicated accessibly (consistent with other call statuses).
- [ ] **Should:** When a user clicks the Keypad button to toggle open the keypad, I would encourage you to have that keypad expand on the page rather than replacing the other elements on the page the way a smartphone interface typically does. It's a lot easier to communicate to screen reader users that a region has been expanded/collapsed than it is to communicate that the whole interface has changed. In terms of keyboard and screen reader interactions, I would want something comparable to the `va-accordion` element, with the `aria-expanded` and `aria-controls` attributes used to communicate state and relationships.
- [ ] **Must:** I don't think we saw the keypad in the prototype provided, but make sure the prototype is keyboard-friendly --- meaning, a user should be able to press the 1 key rather than having to tab to the 1 button and press enter.
- [ ] **Consider:** Does the VA logo in the center of the page provide any benefit to the users? I ask that sincerely because I could imagine it going either way --- maybe it helps build confidence that callers are getting in touch with a trusted resource, or maybe it's just decorative. If it is just decorative, consider removing it. Both it and the primary buttons are circles, and since most of the UI elements here are a departure from the standard VA design system components, it's possible that someone might confuse what's clickable and what's not.
   - [ ] **Must:** If you decide to keep the logo, I think empty `alt` or `aria-hidden="true"` would be appropriate to hide it from screen reader users.
- [ ] **Consider:** There's a chance that some users may land on this page via a route other than the VCL modal, eg. a search result or a direct URL. Consider including text on the screen that directs users to the other VCL options (regular phone, text, chat, TTY, etc) in case one of those is a more appropriate option. I don't think that needs to be super prominent on the page, but maybe something that appears below the calling interface.

## Governance team actions
- [ ] Format feedback as individual tasks (check boxes)
- [ ] Assign this ticket to the VFS team member that opened the Slack request
- [ ] Add the VFS team product label
- [ ] Add the VFS team the feature label (if applicable)
- [ ] Add the touchpoint labels
- [ ] Add the practice area labels
- [ ] Add the Collaboration Cycle initiative milestone

**Comment by briandeconinck** [comment]
Wasn't able to assign to @beccasjames, tagging here for visibility!

cc @artsymartha68 for awareness.

**Comment by beccasjames** [comment]
For "Must: I don't think we saw the keypad in the prototype provided, but make sure the prototype is keyboard-friendly --- meaning, a user should be able to press the 1 key rather than having to tab to the 1 button and press enter." this is the intent of the design/build, but a limitation of Figma's interactions in prototyping mode.
